aten/src/ATen/native/quantized/cpu/qlinear.cpp

```diff
@@ -140,7 +140,11 @@ class QLinearInt8 final : public torch::OperatorKernel {
         /*thread_id=*/0,
         /*num_threads=*/1);
 
-    return output;
+    // The resulting matrix here is 2-D, let's view it with the original
+    // left hand dimensions of the input.
+    std::vector<int64_t> out_sizes = input.sizes().vec();
+    out_sizes.back() = N;
+    return output.view(out_sizes);
```

Comment:
why do you need .view()? just allocate the tensor of the right shape above in _empty_affine_quantized call

Reply:
Thanks for pointing it out. Updated the code to address this issue.